src/Raven.Server/Documents/Expiration/ExpirationStorage.cs

```diff
@@ -145,7 +138,7 @@ public record ExpiredDocumentsOptions
                                             document.TryGetMetadata(out var metadata) == false ||
                                             HasPassed(metadata, metadataPropertyToCheck, options.CurrentTime) == false)
```

Comment:
If the document hasn't expired, we put it in the queue anyway.
We can skip the second `HasPassed` call in the `DeleteDocumentsExpiration` if we don't. Is it possible here?

Reply:
This is the original code. That looks weird.

Reply:
Actually, this is fine.
We are here in the first place because we searched it in the `Expired` tree.
If we cannot find the document or the document isn't expired yet, we need to remove the expiration from the tree.

Reply:
The only way that the document isn't expired is that we moved the expiration of the document.
- The document was saved with expiration 2024.
- The document was modified to expire in 2025.
In the `Expired` tree we'll have 2 entries.

Reply:
Ok, the `modified expiration` case makes the second check essential. And we've already changed `ExpiredDocumentInfo` class name, so it's all right now.